src/bin/cargo.rs

```diff
@@ -309,7 +311,7 @@ fn execute_external_subcommand(config: &Config, cmd: &str, args: &[String]) -> C
     let command = match path {
         Some(command) => command,
         None => {
-            return Err(human(match find_closest(config, cmd) {
+            return Err(CargoError::from(match find_closest(config, cmd) {
```

Comment:
In many cases the compiler was just not figuring it out. I can do another pass to make sure `into` doesn't work. Is there a preference between a turbofished `into` and `from`?